scan/lib/scan/test_command_generator.rb

```diff
@@ -31,8 +31,9 @@ def options
 
         options = []
         options += project_path_array
-        options << "-sdk '#{config[:sdk]}'" if config[:sdk]
-        options << destination # generated in `detect_values`
+        options << "-sdk '#{config[:sdk]}'" if config[:sdk] && !config[:build_for_testing]
+        options << destination unless config[:build_for_testing] # generated in `detect_values`
+        options << "  -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO" if config[:build_for_testing]
```

Comment:
@samskiter correct me if i am wrong but during creating this PR - i discovered that once you set destination you get troubles building for different architectures and ios versions.


so i it ended up to switch from `-destination`   to `-sdk iphonesimulator ONLY_ACTIVE_ARCH=NO`  - found the note somewhere (i think in aws device farm docs)